apps/api-extractor-model/src/model/ApiPackage.ts

```diff
@@ -157,11 +186,35 @@ export class ApiPackage extends ApiItemContainerMixin(ApiNameMixin(ApiDocumented
       }
     }
 
+    const tsdocConfiguration: TSDocConfiguration = new TSDocConfiguration();
+    tsdocConfiguration.clear(true);
+    const { tagDefinitions, supportForTags } = jsonObject.metadata.tsDocConfig;
+    tsdocConfiguration.addTagDefinitions(
+      tagDefinitions.map((definition) => {
+        const { syntaxKind } = definition;
+        const formattedSyntaxKind: TSDocTagSyntaxKind =
```

Comment:
This is the re-mapping I am referring to in my previous comment. Should there be a mechanism from `tsdoc-config` to build up a `TSDocConfigFile` from the product of `TSDocConfigFile.saveToObject()`?

Reply:
@nicholasrice The right solution would be to implement a corresponding `TSDocConfigFile.loadFromObject()` to handle loading of the object. Ideally it would be integrated with the  `TSDocConfigFile.loadFile()` implementation, although there is an interesting nuance that `loadFile()` considers file paths and `"extends"` resolution which assume that the objects have meaningful file paths.  So a bit of abstraction might be needed to handle that.  (Perhaps that is why I procrastinated it?)

If you want to make a PR for TSDoc that would be great.  However your PR has been open for a very long time, so if it helps speed things along, I would also be okay with merging a less ideal solution to get you unblocked. And then we could come back later and improve the `tsdoc-config` API.  

I am away on vacation right now, but I will review this PR as soon as I get back. Sorry about the delays. 

For future reference if you need help getting PRs merged @iclanton's team at Microsoft is a good point of contact, and also we now have a [#contributor-helpline](https://rushstack.zulipchat.com/#narrow/stream/279883-contributor-helpline) chatroom for getting attention of the maintainers.  Thanks!